pkg/goDB/storage/gpfile/gpdir.go

```diff
@@ -190,19 +196,30 @@ func (d *GPDir) ReadBlockAtIndex(colIdx types.ColumnIndex, blockIdx int) ([]byte
 		return nil, ErrDirNotOpen
 	}
 
-	// Load column if required
-	_, err := d.Column(colIdx)
+	// Attempt to read the block
+	data, err := d.readBlockAtIndex(colIdx, blockIdx)
```

Comment:
This will allow recovery even if the write-out happened _after_ the metadata has been read but _before_ a column / `GPFile` has been accessed / lazy-loaded.